src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -0,0 +1,1357 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics;
+
+public class InterceptorsTests : CSharpTestBase
+{
+    private static readonly (string, string) s_attributesSource = ("""
+        namespace System.Runtime.CompilerServices;
+
+        [AttributeUsage(AttributeTargets.Method)]
+        public sealed class InterceptableAttribute : Attribute { }
+
+        [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+        public sealed class InterceptsLocationAttribute : Attribute
+        {
+            public InterceptsLocationAttribute(string filePath, int line, int column)
+            {
+            }
+        }
+        """, "attributes.cs");
+
+    [Fact]
+    public void IsInterceptable()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            class C
+            {
+                [Interceptable]
+                public static void InterceptableMethod() { Console.Write("interceptable"); }
+                
+                public static void NotInterceptable() { Console.Write("not interceptable"); }
+            }
+            """;
+
+        var verifier = CompileAndVerify(new[] { (source, "Program.cs"), s_attributesSource }, symbolValidator: verify, sourceSymbolValidator: verify);
+        verifier.VerifyDiagnostics();
+
+        void verify(ModuleSymbol module)
+        {
+            var method = module.GlobalNamespace.GetMember<MethodSymbol>("C.InterceptableMethod");
+            Assert.True(method.IsInterceptable);
+            Assert.Equal(MethodKind.Ordinary, method.MethodKind);
+
+            method = module.GlobalNamespace.GetMember<MethodSymbol>("C.NotInterceptable");
+            Assert.False(method.IsInterceptable);
```

Comment:
Consider verifying `IsInterceptable` API on each of the different kind of symbol (currently it looks like only source and PE ordinary method cases are directly covered).